### PR TITLE
Add an optional parameter to Envelope which would skip MAC verification

### PR DESCRIPTION
**Pull request by EmilienCourt**
Hi,

As stated in #40, I would like to add an optional parameter, defaulting to `True`, which would allow me to use `dissect.hypervisor` as a library and skip the `cipher.verify` from the Envelope's class.

In my opinion this would not change the upstream code and not introduce any vulnerability, as this defaults to True.

Regards :)

### CODE REVIEW

**Comment by EmilienCourt** [comment]
@DissectBot agree

File: dissect/hypervisor/util/envelope.py [hunk 1]
```diff
@@ -103,7 +103,7 @@
 class Envelope:
     """Implements an encryption envelope as used within ESXi."""
 
-    def __init__(self, fh: BinaryIO):
+    def __init__(self, fh: BinaryIO, should_verify: bool = True):
         self.fh = fh
```
Comment (Schamper) [inline dissect/hypervisor/util/envelope.py:107]:
```suggestion
    def __init__(self, fh: BinaryIO, verify: bool = True):
        self.fh = fh
        self.verify = verify
```

File: dissect/hypervisor/util/envelope.py [hunk 2]
```diff
@@ -126,6 +126,7 @@ def __init__(self, fh: BinaryIO):
         self.key_hash = self.attributes["vmware.keyHash"].value
         self.iv = self.attributes.get("vmware.iv", EnvelopeAttribute(None, None, None)).value
         self.digest = None
+        self.should_verify = should_verify
```
Comment (Schamper) [inline dissect/hypervisor/util/envelope.py:129]:
```suggestion
```

File: dissect/hypervisor/util/envelope.py [hunk 3]
```diff
@@ -190,7 +191,8 @@ def decrypt(self, key: bytes, aad: bytes = None) -> bytes:
         footer = c_envelope.DataTransformCryptoFooter(bytes(decrypted[-512:]))
         decrypted = decrypted[: -4096 - footer.padding]
 
-        cipher.verify(self.digest)
+        if self.should_verify:
```
Comment (Schamper) [inline dissect/hypervisor/util/envelope.py:194]:
```suggestion
        if self.verify:
```

**Review by Schamper — CHANGES_REQUESTED** [review]
Just some small conciseness changes.

**Review by Schamper — APPROVED** [review]

**Pull request merged**